ADBench/plot_graphs.py

```diff
@@ -335,32 +334,39 @@ def values_and_styles(sorted_vals_by_tool):
 
         display_name = utils.format_tool(tool) if len(style) == 2 else style[2]
 
-        yield item, style[0: 2], display_name
+        yield item.values(), style[0: 2], display_name
 
-def generate_graph(figure_info, sorted_vals_by_tool):
-    '''Generates the graph for the given figure.
+def generate_graph(idx, data):
+    '''Generates the graph from the given data.
     
     Args:
-        figure_info (named tuple): information of the figure.
-            idx: index of the figure.
-            build_type: the type of the tool build.
-            objective: the name of the objective, the graph is plotted.
-            maybe_test_size: test size or empty string if the test can not have
-                a size.
-            function_type: type of the current graph (e.g. "Jacobian",
-                "Objective" etc.)
-        sorted_vals_by_tool: values for plotting, sorted by tool name.
+        idx: index of the figure.
+        data (dictionary): plot data and information of the figure.
+            figure_info:
+                build: the type of the tool build.
+                objective: the name of the objective, the graph is plotted.
+                function_type: type of the current graph (e.g. "Jacobian",
+                    "Objective" etc.)
+                test_size: test size or empty string if the test can not have
+                    a size.
+            values:
+                tool: name of the tool.
+                time: array of time values.
+                variable_count: array of variable count, respective to "time".
+                violations: array of bool, determining whether the calculation
+                    was correct or not for each test, respecive to "time".
+                
     '''
 
     # Create figure
-    figure = pyplot.figure(figure_info.idx, figsize=figure_size, dpi=fig_dpi)
+    figure = pyplot.figure(idx, figsize=figure_size, dpi=fig_dpi)
 
     handles, labels = [], []
     non_timeout_violation_x, non_timeout_violation_y = [], []
     additional = []
 
     # Plot results
-    for ((tool, n_vals, t_vals, violations), style, disp_name) in values_and_styles(sorted_vals_by_tool):
+    for ((tool, t_vals, n_vals, violations), style, disp_name) in values_and_styles(data["values"]):
```

Comment:
I think it might be clearer if you return `items` from `values_and_styles` and then explicitly index into that dictionary when you call `label_and_handle` and `together_and_additionals`.

Reply:
Done